back/db/views/analytics_dimension_user_custom_field_values_v01.sql

```diff
@@ -0,0 +1,19 @@
+SELECT
+  DISTINCT u.id as dimension_user_id,
+  cf.key,
+  cf.value
+FROM
+  users u
+  LEFT JOIN LATERAL (
+    SELECT
+      key,
+      custom_field_values ->> key AS value
+    FROM
+      custom_fields
+    where
+      custom_fields.resource_type = 'User'
+  ) cf ON true
+  LEFT JOIN LATERAL (
```

Comment:
Is it important to have NULL custom values in the view? It might be more efficient to not return any nulls if not. When I ran to query I was getting lots of rows with a null value. An inner join on the last query helps, but doesn't eliminate them all.

Reply:
Good question. If we don't add NULL values, then we cannot (or I don't know how to) show how many users didn't specify their custom field (e.g., we show male 20%, female 30%, unspecified 30%, NULL 20%). Currently, we show NULL values on the graphs as `unknown` and I think it makes sense 
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/20be810d-b5e7-4abb-97d8-aeb159478a0c)

Reply:
Ah OK, in that case then it makes sense for the null values to be left in